syntax = "proto3";

package agent_communication.common;

// 通用响应状态
message Status {
    int32 code = 1;
    string message = 2;
    string details = 3;
}

// 服务信息
message ServiceInfo {
    string service_name = 1;
    string version = 2;
    string host = 3;
    int32 port = 4;
    repeated string tags = 5;
    map<string, string> metadata = 6;
}

// 健康检查请求
message HealthCheckRequest {
    string service = 1;
}

// 健康检查响应
message HealthCheckResponse {
    enum ServingStatus {
        UNKNOWN = 0;
        SERVING = 1;
        NOT_SERVING =2;
    }
    ServingStatus status = 1;
}

// 配置信息
message Config {
    string key = 1;
    string value = 2;
    string type = 3;
    int64 timestamp = 4;
}

// 日志级别
enum LogLevel {
    TRACE = 0;
    DEBUG = 1;
    INFO = 2;
    WARN = 3;
    ERROR = 4;
    FATAL = 5;
}

// 日志条目
message LogEntry {
    LogLevel level = 1;
    string message = 2;
    string source = 3;
    int64 timestamp = 4;
    map<string, string> fields = 5;
}